vicodyn/include/cocaine/vicodyn/balancer/utils.hpp

```diff
@@ -5,6 +5,15 @@ namespace cocaine {
 namespace vicodyn {
 namespace balancer {
 
+uint64_t uint64_rand() {
```

Comment:
in the case you truly need random uint64 use modern and flexible <random> library

Reply:
Did you measure it? If no - that's premature optimization again. also you can use LCG instead of MT

Reply:
btw, you can take a look on https://github.com/3Hren/cocaine-plugins/pull/164 
It also mentions rather slow std::uniform_int_distribution, BUT for alternative implementation, which produces much more random numbers generations. In THIS implementation that doesn't count in total time at all.

Reply:
> Did you measure it?

Yes. A thousand times difference.

Reply:
Thank you for the link, I'll study.

Reply:
No, I mean did you measure how does it affects performance in this particular case? btw, `thousand times difference` sounds quite strange, did you use LCG?

Reply:
code:
```c++
#include <bits/stdc++.h>

uint64_t uint64_rand() {
    uint64_t result = std::rand();
    result = (result << 31) | std::rand();
    result = (result << 31) | std::rand();
    return result;
}

uint64_t modern_mt_rand() {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<uint64_t> dis(0, std::numeric_limits<uint64_t>::max());
    return dis(gen);
}

uint64_t modern_mt_rand_st() {
    static std::random_device rd;
    static std::mt19937 gen(rd());
    static std::uniform_int_distribution<uint64_t> dis(0, std::numeric_limits<uint64_t>::max());
    return dis(gen);
}

uint64_t modern_lcg_rand() {
    std::random_device rd;
    std::minstd_rand0 gen(rd());
    std::uniform_int_distribution<uint64_t> dis(0, std::numeric_limits<uint64_t>::max());
    return dis(gen);
}

uint64_t modern_lcg_rand_st() {
    static std::random_device rd;
    static std::minstd_rand0 gen(rd());
    static std::uniform_int_distribution<uint64_t> dis(0, std::numeric_limits<uint64_t>::max());
    return dis(gen);
}

template<class F>
double rps(F f, int try_count) {
    using namespace std::chrono;
    
    auto start = system_clock::now();
    for (int i = 0; i < try_count; ++i) {
        f();
    }
    auto elapsed = system_clock::now()-start;
    auto elapsed_ms = duration_cast<std::chrono::microseconds>(elapsed).count();
    return 1000000. / elapsed_ms * try_count;
}

int main() {
    std::cout << "std::rand          - " << rps(std::rand, 1000000) << " RPS" << std::endl;
    std::cout << "uint64_rand        - " << rps(uint64_rand, 1000000) << " RPS" << std::endl;
    std::cout << "modern_mt_rand     - " << rps(modern_mt_rand, 10000) << " RPS" <<  std::endl;
    std::cout << "modern_mt_rand_st  - " << rps(modern_mt_rand_st, 1000000) << " RPS" <<  std::endl;
    std::cout << "modern_lcg_rand    - " << rps(modern_lcg_rand, 10000) << " RPS" << std::endl;
    std::cout << "modern_lcg_rand_st - " << rps(modern_lcg_rand_st, 1000000) << " RPS" << std::endl;
}
```
output:
```
std::rand          - 7.59821e+07 RPS
uint64_rand        - 2.60539e+07 RPS
modern_mt_rand     - 27266.9 RPS
modern_mt_rand_st  - 4.94047e+07 RPS
modern_lcg_rand    - 34912.3 RPS
modern_lcg_rand_st - 1.90067e+07 RPS
```
My fault was that I recreated generators. Looks like **modern_mt_rand_st** is the best variant for random uint64.